backend/api/views/EventResultView.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from api.models import MeetEvent, Heat
+from api.serializers.EventResultSerializer import EventResultSerializer
+from drf_spectacular.utils import extend_schema, OpenApiParameter
+from api.CustomFilter import filter_by_group
+
+
+
+@extend_schema(tags=['Results'])
+class EventResultView(APIView):
+    @extend_schema(parameters =[OpenApiParameter(name="group_id", type=int)],summary="Displays the results of an event")
+    def get(self, request, event_id):
+        #Get event instance
+        try:
+            event_instance = MeetEvent.objects.get(id=event_id)
+        except MeetEvent.DoesNotExist:
+            return Response({'error': 'Event not found'}, status=status.HTTP_404_NOT_FOUND)
+
+        #Return all heats record with event = even_instance order by heat_time
+        data = self.get_queryset(event_instance)
+        serializer = EventResultSerializer(data, many=True)
+        return Response(serializer.data)
+
+    def get_queryset(self, event_instance):
+        queryset = Heat.objects.filter(event=event_instance)
+        group_id = self.request.query_params.get('group_id')
+        if group_id is not None:
+            queryset = filter_by_group(group_id, queryset, False)
+        return queryset.order_by('heat_time')
```

Comment:
I suggest adding ordering by athlete's last name in case more than one athlete has the same time or status.

Reply:
Thanks for the suggestion to order by the athlete's last name in case of ties—it’s a great point! Now that you’ve mentioned ties, I think we also need to calculate ties explicitly and assign the same place to athletes with the same time or status. This way, the rankings will reflect ties accurately instead of relying solely on ordering

Reply:
I added logic to consider the ties.